src/components/button.component.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  ActivityIndicator,
+  Text,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+import {StyleSheet} from 'react-native';
+
+type Props = TouchableOpacityProps & {
+  text: string;
+  loading?: boolean;
+};
+
+export const ButtonComponent = ({
+  text,
+  loading,
+  ...touchableOpacityProps
+}: Props) => {
+  return (
+    <TouchableOpacity
+      style={styles.button}
+      disabled={loading}
+      {...touchableOpacityProps}>
```

Comment:
era isso mesmo a parte disabled, só tomar cuidado que eu acho que o valor pode ser sobrescrito se for passado, eu acho que vale a pena separar o disabled das touchableOpacityProps e fazer um merge aqui. Mais ou menos isso:

```suggestion
      disabled={disabled  || loading}
      {...touchableOpacityProps}>
```